src/main/scala/com/datastax/gatling/plugin/utils/CqlPreparedStatementUtil.scala

```diff
@@ -9,127 +9,209 @@ package com.datastax.gatling.plugin.utils
 import java.nio.ByteBuffer
+import java.time.{Duration, Instant, LocalDate, LocalTime}
+import java.lang
 import java.util
-import java.util.Date
-import java.util.concurrent.TimeUnit
 
-import com.datastax.driver.core.DataType.Name._
-import com.datastax.driver.core._
-import com.datastax.driver.dse.geometry._
-import com.datastax.driver.dse.geometry.codecs.PointCodec
+import com.datastax.dse.driver.api.core.data.geometry._
+import com.datastax.oss.driver.api.core.cql._
+import com.datastax.oss.driver.api.core.`type`._
+import com.datastax.oss.protocol.internal.ProtocolConstants.DataType._
 import com.datastax.gatling.plugin.exceptions.CqlTypeException
+import com.datastax.oss.driver.api.core.data.{TupleValue, UdtValue}
 import com.github.nscala_time.time.Imports.DateTime
 import io.gatling.core.session.Session
 
 import scala.collection.JavaConverters._
 import scala.util.matching.Regex
 
-
 trait CqlPreparedStatementUtil {
 
   protected val hourMinSecRegEx: Regex = """(\d+):(\d+):(\d+)""".r
   protected val hourMinSecNanoRegEx: Regex = """(\d+):(\d+):(\d+).(\d+{1,9})""".r
 
-  def bindParamByOrder(gatlingSession: Session,
-                       boundStatement: BoundStatement, paramType: DataType.Name,
-                       paramName: String, key: Int): BoundStatement
+  def bindParamByOrder[T <: Bindable[T]](gatlingSession: Session,
+                                         bindable: T,
+                                         paramType: DataType,
+                                         paramName: String,
+                                         key: Int): T
 
-  def bindParamByName(gatlingSession: Session, boundStatement: BoundStatement, paramType: DataType.Name,
-                      paramName: String): BoundStatement
+  def bindParamByName[T <: Bindable[T]](gatlingSession: Session,
+                                        bindable: T,
+                                        paramType: DataType,
+                                        paramName: String): T
 
-  def getParamsMap(preparedStatement: PreparedStatement): Map[String, DataType.Name]
+  def getParamsMap(preparedStatement: PreparedStatement): Map[String, DataType]
 
-  def getParamsList(preparedStatement: PreparedStatement): List[DataType.Name]
+  def getParamsList(preparedStatement: PreparedStatement): List[DataType]
+}
+
+object SessionCollectionResolver {
+  def get(session:Session, name:String):Option[Any] = session.attributes.get(name)
+
+  def getClz[T <: Any](session:Session, name:String):Option[Class[T]] = {
+    get(session,name).flatMap((sessionVal) => {
+      sessionVal match {
+        case clz:Class[T]@unchecked => Option(clz)
+        case _ => Option.empty
+      }
+    })
+  }
+
+  // Note the Java return type here.  This function is used to generate objects which will subsequently
+  // be passed to the Java Bindable impl which is expecting Java Iterators.
+  def getIterable[T <: Any](session:Session, name:String):Option[Iterable[T]] = {
```

Comment:
> Note the Java return type here

Is this comment still up to date? The return type is `Option[scala.Iterable[T]]`.

Reply:
It is not, good catch.  This was changed as a result of the conversation with Tomasz above... and wasn't even accurate in it's original usage.

Fixing now.